ols/src/cache/cache.py

```diff
@@ -40,7 +41,9 @@ def construct_key(user_id: str, conversation_id: str) -> str:
         return f"{user_id}{Cache.COMPOUND_KEY_SEPARATOR}{conversation_id}"
 
     @abstractmethod
-    def get(self, user_id: str, conversation_id: str) -> Union[str, None]:
+    def get(
+        self, user_id: str, conversation_id: str
+    ) -> Union[List[Conversation], None]:
```

Comment:
It is basically `Optional[List[Conversation]]`, but it somehow feels off using the optional type for return, wdyt?

Reply:
`list[Conversation] | None` Looks more clear . find okay @onmete  ?

Reply:
We agreed on using the `Optional` type on Monday's arch meet. I would go with that.
To my understanding, `Optional` has a meaning outside of Python, it basically means when there is nothing meaningful to return, it returns nothing. So it is not wrong to use it for return typing.
Sorry for my original message, I should give it a bit more thought.